src/sbvr-api/odata-response.ts

```diff
@@ -107,11 +109,24 @@ const getFetchProcessingFields = (table: AbstractSqlTable) => {
 		)
 		.map(({ fieldName, dataType }) => {
 			const odataName = sqlNameToODataName(fieldName);
-			return [
-				odataName,
-				(sbvrTypes[dataType as keyof typeof sbvrTypes] as SbvrType)
-					.fetchProcessing,
-			];
+			const sbvrDataType = dataType as keyof typeof sbvrTypes;
+			let fetchProcessingFn = (sbvrTypes[sbvrDataType] as SbvrType)
+				.fetchProcessing;
+
+			// fetchProcessing for WebResources needs to be extended in order
+			// to support presigning a URL immedeately after it is fetched from the database
+			// as no URL can be served without the signing
+			if (sbvrDataType === 'WebResource') {
+				const originalFetchProcessing = fetchProcessingFn;
+				fetchProcessingFn = (data: WebResourceType) => {
```

Comment:
Is there any reason why this is not built in to the data type's `fetchProcessing` itself?

Reply:
@Page- Adding it directly into sbvr-types would require we changed the prototype of `fetchProcessing` to either receive two parameters (the data and the handler) or to have fetchProcessing to receive one parameter data (type `any`) that put together both the webResourceHandler and the data. Either way, it felt that this signing logic did not belong into something that should just handle sbvr types, so that is why I baked into pinejs (and either approach we go it would still require us to check if the type is a WebResource and then forward the webresourceHandler for signing)